Revert "Update template urls (#7638)"

penalosa opened the pull request:
This reverts #7638, which broke links for the Workers Quickstarts page, and in some tutorials.

Fixes https://github.com/cloudflare/workers-sdk/issues/2770

lrapoport-cf commented:
@penalosa i don't think all the links are broken? it may be just that some of the commands need to be fixed -- instead of reverting the whole thing, can we just fix the broken ones?

penalosa commented:
@lrapoport-cf I've reverted the revert (😂) for non-git/wrangler links—how does that look now?

bretthoerner commented:
FWIW I just got this (same issue reported in original bug)

```
npx wrangler generate my-app cloudflare/workers-sdk/tree/main/templates/worker-typescript
Need to install the following packages:
  wrangler
Ok to proceed? (y) 
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
 ⛅️ wrangler 2.11.0 
--------------------
Creating a worker in my-app from cloudflare/workers-sdk/tree/main/templates/worker-typescript

✘ [ERROR] Failed to find "tree/main/templates/worker-typescript" in https://github.com/cloudflare/workers-sdk.git


If you think this is a bug then please create an issue at https://github.com/cloudflare/workers-sdk/issues/new/choose
```

lrapoport-cf commented on content/workers/wrangler-legacy/commands.md:
```diff
@@ -37,7 +37,7 @@ Default values indicated by {{<type>}}=value{{</type>}}.
 
 - `--site` {{<prop-meta>}}optional{{</prop-meta>}}
-  - When defined, the default `$TEMPLATE` value is changed to [`cloudflare/workers-sdk/templates/worker-sites`](https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-sites). This scaffolds a [Workers Site](/workers/platform/sites/) project.
+  - When defined, the default `$TEMPLATE` value is changed to [`cloudflare/worker-sites-template`](https://github.com/cloudflare/worker-sites-template). This scaffolds a [Workers Site](/workers/platform/sites/) project.
```
```suggestion
  - When defined, the default `$TEMPLATE` value is changed to [`cloudflare/workers-sdk/templates/worker-sites`](https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-sites). This scaffolds a [Workers Site](/workers/platform/sites/) project.
```

i think we want to keep the change i had because it points to the version of the worker-sites template that is now in the workers-sdk repo. my understanding is the top-level one is a duplicate and should be deprecated.

lrapoport-cf replied:
@penalosa is there a reason we don't want to reference the worker-sites template in the workers-sdk templates folder?

lrapoport-cf commented on content/workers/wrangler-legacy/commands.md:
```diff
@@ -64,7 +64,7 @@ Default values indicated by {{<type>}}=value{{</type>}}.
 
 - `--site` {{<prop-meta>}}optional{{</prop-meta>}}
-  - When defined, the default `$TEMPLATE` value is changed to [`cloudflare/workers-sdk/templates/worker-sites`](https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-sites). This scaffolds a [Workers Site](/workers/platform/sites/) project.
+  - When defined, the default `$TEMPLATE` value is changed to [`cloudflare/worker-sites-template`](https://github.com/cloudflare/worker-sites-template). This scaffolds a [Workers Site](/workers/platform/sites/) project.
```
```suggestion
  - When defined, the default `$TEMPLATE` value is changed to [`cloudflare/workers-sdk/templates/worker-sites`](https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-sites). This scaffolds a [Workers Site](/workers/platform/sites/) project.
```

as above, i think we want to keep the change i had because it points to the version of the worker-sites template that is now in the workers-sdk repo. my understanding is the top-level one is a duplicate and should be deprecated.

lrapoport-cf replied:
@penalosa is there a reason we don't want to reference the worker-sites template in the workers-sdk templates folder?

lrapoport-cf commented:
> @lrapoport-cf I've reverted the revert (😂) for non-git/wrangler links—how does that look now?

😂 i think there are still some which should be un-reverted, which i've noted :)

penalosa commented on content/workers/get-started/quickstarts.md:
```diff
@@ -50,13 +50,13 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
 ## Templates
 
-{{<worker-starter title="JavaScript Starter" repo="cloudflare/wrangler2/templates/worker" description="A bare-bones Workers starter project, in JavaScript.">}}
+{{<worker-starter title="JavaScript Starter" repo="worker" description="A bare-bones Workers starter project, in JavaScript.">}}
```
The `<worker-starter/>` fragment generates a command of the form `npx wrangler generate my-app $repo`. The value in `repo` is interpreted as either the (bare) name of a Cloudflare template, or a path on the default branch of a repo. Using `tree/main` in the repo path leads to the following command being generated:

```sh
npx wrangler generate my-app cloudflare/workers-sdk/tree/main/templates/worker
```

which results in the following error:

> ✘ [ERROR] Failed to find "tree/main/templates/worker" in https://github.com/cloudflare/workers-sdk.git

If you omit the repo and path, and just use `repo=worker`, Wrangler will look it up in the `workers-sdk` templates folder by default:

```sh
npx wrangler generate my-app worker
```

Alternatively, using `repo=cloudflare/workers-sdk/templates/worker` would also work.

(Fixing these generated commands in the quickstarts was the main purpose of this PR)

penalosa replied:
Oh I _see_, that breaks the links... I think the partial needs to updated for the templates repo move

penalosa commented on content/workers/get-started/quickstarts.md:
```diff
@@ -50,13 +50,13 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
+{{<worker-starter title="TypeScript Starter" repo="worker-typescript" description="A bare-bones Workers starter project, in TypeScript.">}}
 
-{{<worker-starter title="Workers Sites" repo="cloudflare/workers-sdk/tree/main/templates/worker-sites" description="Easily deploy a static site or static assets to Cloudflare’s edge network.">}}
+{{<worker-starter title="Workers Sites" repo="worker-sites" description="Easily deploy a static site or static assets to Cloudflare’s edge network.">}}
```
As above

penalosa commented on content/workers/get-started/quickstarts.md:
```diff
@@ -50,13 +50,13 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
+{{<worker-starter title="Workers Sites" repo="worker-sites" description="Easily deploy a static site or static assets to Cloudflare’s edge network.">}}
 
-{{<worker-starter title="Router" repo="cloudflare/workers-sdk/tree/main/templates/worker-router" description="Run different logic based on the URL and request method. Use this starter to Build REST APIs or apps that require routing logic.">}}
+{{<worker-starter title="Router" repo="worker-router" description="Run different logic based on the URL and request method. Use this starter to Build REST APIs or apps that require routing logic.">}}
```
As above

penalosa commented on content/workers/get-started/quickstarts.md:
```diff
@@ -78,7 +78,7 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
 ## Example Projects
 
-{{<worker-starter title="Speedtest" repo="cloudflare/workers-sdk/tree/main/templates/worker-speedtest"
+{{<worker-starter title="Speedtest" repo="worker-speedtest"
```
As above

lrapoport-cf commented:
@penalosa links all seem to be working now :) just one more followup question -- how was it decided which examples to remove from the quickstarts page? i thought it would track with those in the `templates/experimental` folder, but looks like more than just those were removed from the page. should more of our templates be moved into `templates/experimental`?

penalosa commented:
- I removed the TypeScript & Javascript Starter since they’re covered by  `wrangler init`
- I removed the Workers Sites starter & Miniflare Example Project since we don’t recommend them for new users (covered by Pages and Wrangler respectively)
- I moved Sundar & Hono to the Frameworks section

lrapoport-cf commented on content/workers/get-started/quickstarts.md:
```diff
@@ -89,43 +64,27 @@ description="Measure download / upload connection speed from the client side, us
+## Frameworks
 
-{{<worker-starter title="Hello World (Kotlin)" repo="cloudflare/kotlin-worker-hello-world" description="A bare-bones starter in Kotlin.">}}
+{{<worker-starter title="Hono Starter" repo="honojs/hono-minimal" description="Hono is an ultrafast web framework built for Cloudflare Workers. This is a minimal project using Hono, TypeScript, esbuild, Miniflare, and Jest.">}}
```
suggestion to remove "starter" to be consistent with the format of the rest of the titles in this section
```suggestion
{{<worker-starter title="Hono" repo="honojs/hono-minimal" description="Hono is an ultrafast web framework built for Cloudflare Workers. This is a minimal project using Hono, TypeScript, esbuild, Miniflare, and Jest.">}}
```

lrapoport-cf commented on content/workers/get-started/quickstarts.md:
```diff
@@ -89,43 +64,27 @@ description="Measure download / upload connection speed from the client side, us
-{{<worker-starter title="AWS DynamoDB SQS" repo="cloudflare/workers-sdk/tree/main/templates/worker-aws" description="Use AWS services such as DynamoDB and SQS from a Cloudflare Worker">}}
+{{<worker-starter title="AWS DynamoDB SQS" repo="cloudflare/workers-sdk/templates/worker-aws" description="Use AWS services such as DynamoDB and SQS from a Cloudflare Worker">}}
 
 {{<worker-starter title="Edge-side rendering - Vitedge" repo="frandiox/vitessedge-template" description="Use Vite to render pages at the edge with great DX. Includes i18n, markdown support and more.">}}
```
let's capitalize for consistency
```suggestion
{{<worker-starter title="Edge-Side Rendering - Vitedge" repo="frandiox/vitessedge-template" description="Use Vite to render pages at the edge with great DX. Includes i18n, markdown support and more.">}}
```

deadlypants1973 replied:
```suggestion
{{<worker-starter title="Edge-Side Rendering - Vitedge" repo="frandiox/vitessedge-template" description="Use Vite to render pages at the edge with great developer experience (DX). Includes i18n, markdown support and more.">}}
```

lrapoport-cf commented:
> * I removed the TypeScript & Javascript Starter since they’re covered by  `wrangler init`
> * I removed the Workers Sites starter & Miniflare Example Project since we don’t recommend them for new users (covered by Pages and Wrangler respectively)
> * I moved Sundar & Hono to the Frameworks section

@penalosa  👍  

as a followup PR to workers-sdk, does it make sense to move `worker-rust`, `worker-cobol`, and `worker-emscripten` to the `templates/experimental` folder?

lrapoport-cf commented:
@penalosa i added two more small nit suggestions. just have an outstanding question re: worker-sites (https://github.com/cloudflare/cloudflare-docs/pull/7728#discussion_r1116308466 and https://github.com/cloudflare/cloudflare-docs/pull/7728#discussion_r1116308466

deadlypants1973 commented on content/workers/get-started/quickstarts.md:
```diff
@@ -48,37 +47,13 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
 ## Example Projects
 
-{{<worker-starter title="Speedtest" repo="cloudflare/workers-sdk/tree/main/templates/worker-speedtest"
+{{<worker-starter title="Router" repo="cloudflare/workers-sdk/templates/worker-router" description="Run different logic based on the URL and request method. Use this starter to Build REST APIs or apps that require routing logic.">}}
```
```suggestion
{{<worker-starter title="Router" repo="cloudflare/workers-sdk/templates/worker-router" description="Run different logic based on the URL and request method. Use this starter to Build REST APIs or applications that require routing logic.">}}
```

deadlypants1973 commented on content/workers/get-started/quickstarts.md:
```diff
@@ -89,43 +64,27 @@ description="Measure download / upload connection speed from the client side, us
 {{<worker-starter title="BinAST" repo="xtuc/binast-cf-worker-template" description="Serve a JavaScript Binary AST via a Cloudflare Worker.">}}
 
-{{<worker-starter title="AWS DynamoDB SQS" repo="cloudflare/workers-sdk/tree/main/templates/worker-aws" description="Use AWS services such as DynamoDB and SQS from a Cloudflare Worker">}}
+{{<worker-starter title="AWS DynamoDB SQS" repo="cloudflare/workers-sdk/templates/worker-aws" description="Use AWS services such as DynamoDB and SQS from a Cloudflare Worker">}}
```
```suggestion
{{<worker-starter title="AWS DynamoDB SQS" repo="cloudflare/workers-sdk/templates/worker-aws" description="Use AWS services such as DynamoDB and SQS from a Cloudflare Worker.">}}
```

deadlypants1973 commented on content/workers/get-started/quickstarts.md:
```diff
@@ -48,37 +47,13 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
+{{<worker-starter title="Speedtest" repo="cloudflare/workers-sdk/templates/worker-speedtest"
 description="Measure download / upload connection speed from the client side, using the Performance Timing API.">}}
 
 {{<worker-starter title="Sentry" repo="bustle/cf-sentry" description="Log exceptions and errors in your Workers application to Sentry.io - an error tracking tool">}}
```
```suggestion
{{<worker-starter title="Sentry" repo="bustle/cf-sentry" description="Log exceptions and errors in your Workers application to Sentry.io - an error tracking tool.">}}
```

deadlypants1973 commented on content/workers/get-started/quickstarts.md:
```diff
@@ -89,43 +64,27 @@ description="Measure download / upload connection speed from the client side, us
+{{<worker-starter title="Hono Starter" repo="honojs/hono-minimal" description="Hono is an ultrafast web framework built for Cloudflare Workers. This is a minimal project using Hono, TypeScript, esbuild, Miniflare, and Jest.">}}
 
-{{<worker-starter title="Hello World (COBOL)" repo="cloudflare/workers-sdk/tree/main/templates/worker-cobol" description="A bare-bones starter in COBOL.">}}
+{{<worker-starter title="Apollo GraphQL Server" repo="signalnerve/workers-graphql-server" description="Lightning-fast, globally distributed Apollo GraphQL server, deployed at the edge using Cloudflare Workers.">}}
```
```suggestion
{{<worker-starter title="Apollo GraphQL Server" repo="signalnerve/workers-graphql-server" description="Lightning-fast, globally distributed Apollo GraphQL server, deployed on the Cloudflare global network using Cloudflare Workers.">}}
```

deadlypants1973 reviewed (approved):
PCX reviewed with comment. We are not using the word 'edge' to describe the CF network anymore and are now using 'Cloudflare global network'. Also added in some periods at the end of descriptions.

penalosa commented:
@lrapoport-cf & @deadlypants1973 those should all be addressed now.

> as a followup PR to workers-sdk, does it make sense to move worker-rust, worker-cobol, and worker-emscripten to the templates/experimental folder?

Yes, we should definitely do that

lrapoport-cf commented on content/workers/get-started/quickstarts.md:
```diff
@@ -48,84 +47,44 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
+description="Measure download/upload connection speed from the client side, using the Performance Timing API.">}}
 
-{{<worker-starter title="Sentry" repo="bustle/cf-sentry" description="Log exceptions and errors in your Workers application to Sentry.io - an error tracking tool">}}
+{{<worker-starter title="Sentry" repo="bustle/cf-sentry" description="Log exceptions and errors in your Workers application to Sentry.io - an error tracking tool.">}}
```
```suggestion
{{<worker-starter title="Sentry" repo="mhart/cf-sentry" description="Log exceptions and errors in your Workers application to Sentry.io - an error tracking tool.">}}
```

`bustle/cf-sentry` redirects to `mhart/cfsentry`

lrapoport-cf commented on content/workers/get-started/quickstarts.md:
```diff
@@ -48,84 +47,44 @@ $ npx wrangler generate <new-project-name> <github-repo-url>
+{{<worker-starter title="Hono" repo="honojs/hono-minimal" description="Hono is an ultrafast web framework built for Cloudflare Workers. This is a minimal project using Hono, TypeScript, esbuild, Miniflare, and Jest.">}}
 
-{{<worker-starter title="Hello World (COBOL)" repo="cloudflare/workers-sdk/tree/main/templates/worker-cobol" description="A bare-bones starter in COBOL.">}}
+{{<worker-starter title="Apollo GraphQL Server" repo="signalnerve/workers-graphql-server" description="Lightning-fast, globally distributed Apollo GraphQL server, deployed on the Cloudflare global network using Cloudflare Workers.">}}
```
this redirects to https://github.com/cloudflare/workers-graphql-server -- can we update the `repo=` param so that the command displayed reflects this?

lrapoport-cf commented:
@penalosa i found two other small things that would be great to fix but otherwise lgtm :) thank you!

penalosa commented:
@lrapoport-cf That should (🤞) be everything now

lrapoport-cf commented:
> @lrapoport-cf & @deadlypants1973 those should all be addressed now.
> 
> > as a followup PR to workers-sdk, does it make sense to move worker-rust, worker-cobol, and worker-emscripten to the templates/experimental folder?
> 
> Yes, we should definitely do that

created https://github.com/cloudflare/workers-sdk/issues/2846

lrapoport-cf commented:
> @lrapoport-cf That should (🤞) be everything now

looks good!
